src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/PSUserAgent.cs

```diff
@@ -131,7 +132,8 @@ internal static string PlatformName
                     // only generate the windows user agent once
                     if(s_windowsUserAgent == null){
                         // find the version in the windows operating system description
-                        string versionText = OS.Substring(OS.LastIndexOf(" ") +1);
+                        Regex pattern = new Regex(@"\d+(\.\d+)+");
+                        string versionText = pattern.Match(OS).Value;
                         Version windowsPlatformversion = new Version(versionText);
                         s_windowsUserAgent = $"Windows NT {windowsPlatformversion.Major}.{windowsPlatformversion.Minor}";
```

Comment:
Why do we parse? Maybe just return  OS (OSDescription) ?

Reply:
Because of the format expected in User-Agents. the OS portion should be `Windows NT <Major Version>.<Minor Version>` That's not native anywhere in the CoreFX name space, `System.Environment.OSVersion.Version` reports `6.2` for major/minor on windows 10, but User-Agent expects `10.0`. ugh.

Reply:
Please clarify for me what is "the format expected in User-Agents"? My understanding is that the string is sent to a web server as UserAgent string and falls in a web log file so if I see the `OSDescription` in the log it is useful.

Reply:
User-Agents have an expected format. The first part, Platform, has a specific format to detect Windows/Linux/macOS. the full `OSDescription` follows the Platform field. here is the full user-agent strings with this change:

windows 10:
```none
Mozilla/5.0 (Windows NT 10.0; Microsoft Windows 10.0.15063; en-US) PowerShell/6.0.0
```

windows 7
```none
Mozilla/5.0 (Windows NT 6.1; Microsoft Windows 6.1.7601 S; en-US) PowerShell/6.0.0
```

The change in this PR is for the `Windows NT 10.0` and `Windows NT 6.1` Platform fields.

Reply:
@markekraus I think we must merge and fix the bug and I'll open new Issue to discuss User-Agents.